### PR TITLE
Separated game path configurations and swapped RCT2_ACCESS with global configuration.

### PR DESCRIPTION
**Pull request by xiane**
- Separated game path configuration (game_path, game_path_slash ..etc) from gConfigGeneral structure.

- Created to initialise directories information what separated.

- Swapped RCT2_ACCESS with global configurations.

- And questions : what is RCT2_ADDRESS_SAVED_GAMES_PATH_2? it not be used saving saved game path, just used like.. buffer? it just be coped one time! what is major use of it? please let me know it.

- Please somebody test it in other environment. it just be tested in OS X.

### CODE REVIEW

File: src/config.h [hunk 1]
```diff
@@ -132,12 +132,21 @@ enum {
 	SCENARIO_SELECT_MODE_DIFFICULTY,
 	SCENARIO_SELECT_MODE_ORIGIN,
 };
+typedef struct {
+	utf8string game_path;
+	utf8string game_path_slash;
```
Comment (IntelOrca) [inline src/config.h]: This is dynamically resolved and should not be part of the config.

File: src/config.h [hunk 2]
```diff
@@ -132,12 +132,21 @@ enum {
...
+	utf8string saved_game_path;
+	utf8string scenario_path;
+	utf8string landscapes_path;
+	utf8string object_data_path;
+	utf8string tracks_path;
+	utf8string saved_game_path_2;
```
Comment (IntelOrca) [inline src/config.h]: I think this is legacy of the original game and doesn't need to be used any more. I am not sure what was different about it.

**Comment by IntelOrca** [comment]
Overall, whilst I am fine with the paths being stored for each sub directory, I don't see why they all need to be in the config file.

**Comment by Gymnasiast** [comment]
To me, this doesn't seem to be very beneficial, but it does introduce a lot of additional code complexity. Also, having one game_path variable seems to be borderline complex for many novice users, let alone several.

**Comment by xiane** [comment]
Hmm.. Do you means.. It doesn't need to be configuration And create string of subdirectories dynamically every start time or need?
Well,  How about create global subdirectories list and rename game_path to root_directory?? I think matching means and name make more good.

File: src/config.h (hunk 1 above)
Comment (xiane) [inline src/config.h]: Okay. I'm going to apply it

**Comment by IntelOrca** [comment]
@xiane what I think you should do is remove all the variables you added to config, and instead create a new function called get_rct2_directory which behaves like platform_get_user_directory. You pass it the buffer to write to and the sub directory you want. It sets the given buffer to rct2 path \ sub directory.

For example:
```c
enumFileHandle = platform_enumerate_files_begin(gConfigGamePath.tracks);
```

would become

```c
utf8 path[MAX_PATH];
get_rct2_directory(path, "tracks");
strncat_path(path, "*.td6", sizeof(path));
enumFileHandle = platform_enumerate_files_begin(path);
```

**Comment by xiane** [comment]
Okay, so i stop this PR.after that, applying your answer and do re PR. thank you!

**Pull request closed without merging**